### PR TITLE
Rename S32 to NXP S32

### PR DESCRIPTION
**Pull request by manuargue**
Following updates previously done for other drivers and as requested by reviewers (e.g. https://github.com/zephyrproject-rtos/zephyr/pull/51451#issuecomment-1294384579), rename all occurrences of S32 to NXP S32 to avoid ambiguity.

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]
LGTM once the HAL change is merged.

**Review by dcpleung — DISMISSED** [review]

**Review by dleach02 — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**
